src/lib/Libifl/pbsD_connect.c

```diff
@@ -1146,7 +1149,7 @@ pbs_register_sched(const char *sched_id, int primary_conn_id, int secondary_conn
 	svr_conn_t **svr_conns_primary = NULL;
 	svr_conn_t **svr_conns_secondary = NULL;
 
-	if (sched_id == NULL)
```

Comment:
These checks are redundant after the changes when get_conn_svr_instances() returns NULL. Following lines make the same check.

Reply:
I think the checks are fine. There is no point in calling get_conn_svr_instances when the connection fd itself is invalid.

Reply:
In that analogy, the scheduler should not have called pbs_register_sched() in case if either of the fd is negative. Once you are here there is no need for redundant check in consecutive lines.
Also Shouldn't this function be inside src/scheduler?

Reply:
Since this is part of the library that any client can use, It is not wise to rely on the client to do the right thing before calling the routine. I think this check is fine.
Your second comment is related to - https://github.com/openpbs/openpbs/pull/2068#issuecomment-706158276
The comment mentions the reason why it was done so.

Reply:
i think there's no harm in having a function do its own checking, it's good coding practice. What's the harm?